lib/template-parts.php

```diff
@@ -213,6 +213,17 @@ function filter_rest_wp_template_part_query( $args, $request ) {
 			'value' => $request['theme'],
 		);
 
+		// Ensure auto-drafts of all theme supplied template parts are created.
+		if( $request['theme'] === get_current_theme() ) {
+			foreach ( get_template_types() as $template_type ) {
+				// Skip 'embed' for now because it is not a regular template type.
+				if ( in_array( $template_type, array( 'embed' ), true ) ) {
+					continue;
+				}
+				gutenberg_find_template_post_and_parts( $template_type );
+			}
```

Comment:
I think this is the approach @ockham and I discussed in slack.  Essentially since `gutenberg_find_template_post_and_parts` handles locating the block template files and creating the template part auto drafts, we can pass the possible template types into there to save auto-draft template parts for the theme.

Reply:
Why aren't theme-supplied template parts turned into auto-drafts as part of the normal site editor loading? (i kinda thought they were)

Reply:
This won't work. That function only creates auto drafts for template parts used in the resolved template of the passed in type.

We need to create auto drafts for all files in the template-parts folder.

Reply:
If we are doing this for the entire list of template types, and template parts belong to those templates before they are changed... then aren't auto-drafts of the template part created before there is a chance for them to not be part of the resolved template?

Reply:
> Why aren't theme-supplied template parts turned into auto-drafts as part of the normal site editor loading? (i kinda thought they were)

Good question?

Reply:
> We need to create auto drafts for all files in the template-parts folder.

So maybe more along the lines of:
```
if( $request['theme'] === get_current_theme() ) {
	$template_part_files = glob( get_stylesheet_directory() . '/block-template-parts/*.html' );
	$template_part_files = is_array( $template_part_files ) ? $template_part_files : array();
	foreach ( $template_part_files as $template_part ) {
		create_auto_draft_for_template_part_block( $template_part );
	}
};
```
?

Reply:
Yes, because themes can provide template parts not used in any resolved templates that are there just as alternatives.

That works, but you need to account for child themes like the template code does.

Reply:
Just pushed that with the child theme step. i was going to ask about that. Thanks 😄 

> Yes, because themes can provide template parts not used in any resolved templates that are there just as alternatives.

Ok, gotcha. that makes sense.

Reply:
Im wondering if there would be a better place to put this?  Would it be called too often in the rest filter?  Would it make more sense to do this on load or anywhere else? 🤔

Reply:
We should include a check so that it looks for existing auto drafts before duplicating them. The template code also does that.

Reply:
> We should include a check so that it looks for existing auto drafts before duplicating them. The template code also does that.

Looking further, isn't that already handled in the `create_auto_draft_for_template_part_block` function?
https://github.com/WordPress/gutenberg/blob/881b7f22c1aa472b09a5e9b02b2282b5e5f97e05/lib/template-loader.php#L152-L167

Or do we need to not use this function, and do a similar check but only for 'auto-draft' status so that the originals are still available even after customized versions have been saved?

Reply:
Ah, no, this is fine.